pkg/controller/resourcequota/replenishment_controller.go

```diff
@@ -87,42 +88,44 @@ func ObjectReplenishmentDeleteFunc(options *ReplenishmentControllerOptions) func
 // ReplenishmentControllerFactory knows how to build replenishment controllers
 type ReplenishmentControllerFactory interface {
 	// NewController returns a controller configured with the specified options
-	NewController(options *ReplenishmentControllerOptions) (*framework.Controller, error)
+	NewController(options *ReplenishmentControllerOptions) (framework.ControllerInterface, error)
 }
 
 // replenishmentControllerFactory implements ReplenishmentControllerFactory
 type replenishmentControllerFactory struct {
-	kubeClient clientset.Interface
+	kubeClient  clientset.Interface
+	podInformer framework.SharedInformer
 }
 
 // NewReplenishmentControllerFactory returns a factory that knows how to build controllers
 // to replenish resources when updated or deleted
-func NewReplenishmentControllerFactory(kubeClient clientset.Interface) ReplenishmentControllerFactory {
+func NewReplenishmentControllerFactory(podInformer framework.SharedInformer, kubeClient clientset.Interface) ReplenishmentControllerFactory {
```

Comment:
Ok,will hold you to that :-)

On Tuesday, April 19, 2016, David Eads <notifications@github.com> wrote:

> In pkg/controller/resourcequota/replenishment_controller.go
> <https://github.com/kubernetes/kubernetes/pull/24470#discussion_r60270619>
> :
>
> >  }
> >
> >  // NewReplenishmentControllerFactory returns a factory that knows how to build controllers
> >  // to replenish resources when updated or deleted
> > -func NewReplenishmentControllerFactory(kubeClient clientset.Interface) ReplenishmentControllerFactory {
> > +func NewReplenishmentControllerFactory(podInformer framework.SharedInformer, kubeClient clientset.Interface) ReplenishmentControllerFactory {
>
> at what point can we bite the bullet and have a informer registry tied to
> a particular type of resource so we can avoid passing in multiple informers
> in this operation?
>
> Probably two.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/24470/files/1bf310124e5d41b602cef6110cf70942036c5140#r60270619>
>